httpie/utils.py

```diff
@@ -12,8 +12,46 @@
 RE_COOKIE_SPLIT = re.compile(r', (?=[^ ;]+=)')
 
 
-def load_json_preserve_order(s):
-    return json.loads(s, object_pairs_hook=OrderedDict)
+class JsonDictPreservingDuplicateKeys(OrderedDict):
```

Comment:
I will use `dict` parent class and add a custom `__repr__()`. It should be cleaner.